File: c/Selectsort.c
#include <stdio.h>

void selectsort(int a[], int n){
    int i, j, temp;
    for(i = 0; i < n; i++){
        for(j = i + 1; j <= n; j++){
            if(a[i] > a[j]){
                temp = a[i];
                a[i] = a[j];
                a[j] = temp;
            }
        }
    }
}

int main(){
    int v[] = {2,3,4,52,23,0,9,7,5};
    int len = sizeof(v)/sizeof(int);
    selectsort(v,len);
    for(int i = 0; i < len; i++){
        printf("%d ", v[i]);
    }
}

File: golang/SelectSort.go
package main

import "fmt"

func SelectSort(vector []int) {
	fmt.Println("SelectSort")
	fmt.Println(vector)
	for i := 0; i < len(vector); i++ {
		// 选择最小的元素
		k := i
		for j := i + 1; j < len(vector); j++ {
			if vector[k] > vector[j] {
				k = j
			}
		}
		// 交换
		if k != i {
			temp := vector[i]
			vector[i] = vector[k]
			vector[k] = temp
		}
		fmt.Println(vector)
	}
}

func main() {
	SelectSort([]int{3, 4, 2, 4, 6, 7, 8, 4})
}
